include/robot_state_publisher/robot_state_publisher.hpp

```diff
@@ -34,72 +34,85 @@
 
 /* Author: Wim Meeussen */
 
-#ifndef ROBOT_STATE_PUBLISHER__ROBOT_STATE_PUBLISHER_H_
-#define ROBOT_STATE_PUBLISHER__ROBOT_STATE_PUBLISHER_H_
-
+#ifndef ROBOT_STATE_PUBLISHER__ROBOT_STATE_PUBLISHER_HPP_
+#define ROBOT_STATE_PUBLISHER__ROBOT_STATE_PUBLISHER_HPP_
+
+#include <builtin_interfaces/msg/time.hpp>
+#include <kdl/tree.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/joint_state.hpp>
+#include <std_msgs/msg/string.hpp>
+#include <tf2_ros/static_transform_broadcaster.h>
+#include <tf2_ros/transform_broadcaster.h>
+#include <urdf/model.h>
+
+#include <chrono>
 #include <map>
+#include <memory>
 #include <string>
+#include <vector>
 
-#include "kdl/frames.hpp"
-#include "kdl/segment.hpp"
-#include "kdl/tree.hpp"
-
-#include "tf2_ros/static_transform_broadcaster.h"
-#include "tf2_ros/transform_broadcaster.h"
-#include "urdf/model.h"
-
-#include "std_msgs/msg/string.hpp"
-
-#include "rclcpp/rclcpp.hpp"
+typedef std::map<std::string, urdf::JointMimicSharedPtr> MimicMap;
 
 namespace robot_state_publisher
 {
 
-class SegmentPair
+class SegmentPair final
 {
 public:
-  SegmentPair(const KDL::Segment & p_segment, const std::string & p_root, const std::string & p_tip)
+  explicit SegmentPair(
+    const KDL::Segment & p_segment,
+    const std::string & p_root,
+    const std::string & p_tip)
   : segment(p_segment), root(p_root), tip(p_tip) {}
 
   KDL::Segment segment;
-  std::string root, tip;
+  std::string root;
+  std::string tip;
 };
 
-class RobotStatePublisher
+class RobotStatePublisher final : public rclcpp::Node
 {
 public:
-  /** Constructor
-   * \param tree The kinematic model of a robot, represented by a KDL Tree
-   */
-  RobotStatePublisher(
-    rclcpp::Node::SharedPtr node_handle, const KDL::Tree & tree,
-    const urdf::Model & model, const std::string & model_xml);
+  /// Constructor
+  explicit RobotStatePublisher(const rclcpp::NodeOptions & options);
 
   /// Destructor
-  virtual ~RobotStatePublisher() {}
+  ~RobotStatePublisher();
```

Comment:
Doesn't look like this does anything so ` = default` would be safe.

Reply:
Sure.